sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1074,6 +1089,8 @@ def test_valid_brute_metric_for_auto_algorithm():
 
     for metric in VALID_METRICS_SPARSE['brute']:
         if metric != 'precomputed' and metric not in require_params:
+            if metric == 'haversine':
```

Comment:
Should haversine be in `VALID_METRICS_SPARSE['brute']`?

Reply:
I added this line